tests/data/test_types.py

```diff
@@ -157,13 +178,41 @@ def test_save_trajectories(self, trajectory_rew, use_chdir, tmpdir):
             chdir_context = contextlib.nullcontext()
             save_dir = tmpdir
 
-        trajs = [trajectory_rew]
+        if not use_rewards:
+            trajs = [_remove_reward(trajectory_rew)]
+        else:
+            trajs = [trajectory_rew]
+
         save_path = pathlib.Path(save_dir, "trajs.pkl")
 
         with chdir_context:
-            types.save(str(save_path), trajs)
-            with open(save_path, "rb") as f:
-                loaded_trajs = pickle.load(f)
+            if use_pickle:
+                # Pickle format
+                with open(save_path, "wb") as f:
+                    pickle.dump(trajs, f)
+            else:
+                # .npz format
+                types.save(save_path, trajs)
+
+                # Test that heterogeneous lists of trajectories throw an error
+                if use_rewards:
+                    trajs_bad = [
+                        _remove_reward(trajectory_rew),
+                        trajectory_rew,
+                    ]
+                    with pytest.raises(ValueError):
+                        types.save(save_path, trajs_bad)
+
+            if type_safe:
+                if use_rewards:
+                    loaded_trajs = types.load_with_rewards(save_path)
+                else:
+                    with pytest.raises(ValueError):
+                        loaded_trajs = types.load_with_rewards(save_path)
+                    loaded_trajs = types.load(save_path)
+            else:
+                loaded_trajs = types.load(save_path)
+
             assert len(trajs) == len(loaded_trajs)
             for t1, t2 in zip(trajs, loaded_trajs):
                 _assert_dataclasses_equal(t1, t2)
```

Comment:
Note this only passes because `trajectory_rew` includes non-empty `infos`, otherwise it would fail the round-trip (None turns into `[{}, ..., {}]`).

This is fairly innocuous though and doesn't seem to even add much space overhead, but you might want to at least document it in `save`.

Reply:
Now documented